### PR TITLE
[move-spec-test] report generation

### PR DESCRIPTION
**Pull request by asmie**
### Description
This PR adds basic report generation. Report is generated per original source file and can be saved to the file in JSON form (new switch `-o`). JSON report looks like:
```json
{
  "files": {
    "./sources/m1/Negation.move": {
      "tested": 1,
      "killed": 1
    },
    "./sources/m1/m1_1/Negation.move": {
      "tested": 1,
      "killed": 1
    },
    "./sources/m2/Negation.move": {
      "tested": 1,
      "killed": 1
    },
    "./sources/Negation.move": {
      "tested": 1,
      "killed": 1
    }
  }
}
```

Summary is also displayed to the screen in tabelarized form.
Example:
```bash
Total mutants tested: 4
Total mutants killed: 4

╭─────────────────────────────────┬────────────────┬────────────────┬────────────╮
│ File                            │ Mutants tested │ Mutants killed │ Percentage │
├─────────────────────────────────┼────────────────┼────────────────┼────────────┤
│ ./sources/m2/Negation.move      │ 1              │ 1              │ 100.00%    │
├─────────────────────────────────┼────────────────┼────────────────┼────────────┤
│ ./sources/m1/Negation.move      │ 1              │ 1              │ 100.00%    │
├─────────────────────────────────┼────────────────┼────────────────┼────────────┤
│ ./sources/Negation.move         │ 1              │ 1              │ 100.00%    │
├─────────────────────────────────┼────────────────┼────────────────┼────────────┤
│ ./sources/m1/m1_1/Negation.move │ 1              │ 1              │ 100.00%    │
╰─────────────────────────────────┴────────────────┴────────────────┴────────────╯
```

### Test Plan
New unit tests added for the reporting module.

### CODE REVIEW

File: third_party/move/tools/move-spec-test/src/report.rs [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+use std::path::PathBuf;
+use tabled::{builder::Builder, settings::Style};
+
+/// This struct represents a report of the specification testing.
+/// It contains the list of entries, where each entry is a file and the number of mutants tested
+/// and killed in that file (in form of a `ReportEntry` structure).
+#[derive(Debug, Serialize)]
+pub struct Report {
+    /// The list of entries in the report.
+    files: HashMap<PathBuf, MutantStats>,
```
Comment (vineethk) [inline third_party/move/tools/move-spec-test/src/report.rs:12]: Might be good to use a deterministic datastructure like `BTreeMap` for reproducibility of the report (fixed order for entries).

File: third_party/move/tools/move-spec-test/src/report.rs [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+        let entry = self
+            .files
+            .entry(path.clone())
+            .or_insert(MutantStats::default());
+        entry.tested += 1;
+    }
+
+    /// Increments the number of mutants killed for the given path.
+    /// If the path is not in the report, it adds it with the number of mutants tested and killed
+    /// set to the default (0) and then increases only killed count!
```
Comment (vineethk) [inline third_party/move/tools/move-spec-test/src/report.rs:35]:
```suggestion
    /// count set to 1.
```

File: third_party/move/tools/move-spec-test/src/report.rs [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+    /// If the path is not in the report, it adds it with the number of mutants tested set to 1.
+    pub fn increment_mutants_tested(&mut self, path: &PathBuf) {
+        let entry = self
+            .files
+            .entry(path.clone())
+            .or_insert(MutantStats::default());
+        entry.tested += 1;
+    }
+
+    /// Increments the number of mutants killed for the given path.
```
Comment (vineethk) [inline third_party/move/tools/move-spec-test/src/report.rs:33]:
```suggestion
    /// Increments the number of mutants killed for the given path by 1.
```

**Review by vineethk — COMMENTED** [review]
lgtm with some comments

File: third_party/move/tools/move-spec-test/src/report.rs (hunk 2 above)
Comment (asmie) [reply third_party/move/tools/move-spec-test/src/report.rs:35]: Done in #86

File: third_party/move/tools/move-spec-test/src/report.rs (hunk 3 above)
Comment (asmie) [reply third_party/move/tools/move-spec-test/src/report.rs:33]: > Increments

The comment begins with "increment, " which means increase by 1. I'm unsure if there is a need to explicitly add a "by 1" phrase, as incrementing cannot be done with other counts.

However, if it's hard to catch, I can rephrase it to make it more obvious.

File: third_party/move/tools/move-spec-test/src/report.rs (hunk 1 above)
Comment (asmie) [reply third_party/move/tools/move-spec-test/src/report.rs:12]: Done in #86

File: third_party/move/tools/move-spec-test/src/report.rs (hunk 3 above)
Comment (vineethk) [reply third_party/move/tools/move-spec-test/src/report.rs:33]: This was a minor suggestion, but "increment" just means increase or addition, the quantity is not implicit (https://www.dictionary.com/browse/increment). You can certainly increment by any count.
